src/cocotb/_py_compat.py

```diff
@@ -84,30 +84,13 @@ def __exit__(self, *excinfo: object) -> None:
     from functools import cached_property
 else:
     from functools import update_wrapper
-    from typing import Any, Callable, Generic, Type, TypeVar, Union, overload
 
-    T_co = TypeVar("T_co", covariant=True)
-
-    class cached_property(Generic[T_co]):
-        def __init__(self, method: Callable[..., T_co]) -> None:
+    class cached_property:
+        def __init__(self, method):
             self._method = method
-            update_wrapper(self, method)  # type: ignore[arg-type]
```

Comment:
Just for history: the diff looks strange here, the update_wrapper() call is only modified further down, it's not removed.